File: Baekjoon/3986.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
	int n;
	cin >> n;
	int answer = 0;
	while (n--) {
		string s;
		cin >> s;
		stack<char> st;
		for (char c : s) {
			if (st.empty() || st.top() != c) st.push(c);
			else st.pop();
		}
		if (st.empty()) answer++;
	}
	cout << answer << '\n';
	return 0;
}

File: Baekjoon/3986.py
n = int(input())

answer = 0
for _ in range(n):
    s = input()
    stack = []
    for i in range(len(s)):
        if len(stack) == 0 or stack[-1] != s[i]:
            stack.append(s[i])
        else:
            stack.pop()
    if len(stack) == 0:
        answer += 1
print(answer)
